channeldb/payment_control.go

```diff
@@ -615,19 +619,45 @@ func fetchPaymentBucketUpdate(tx kvdb.RwTx, paymentHash lntypes.Hash) (
 
 // nextPaymentSequence returns the next sequence number to store for a new
 // payment.
-func nextPaymentSequence(tx kvdb.RwTx) ([]byte, error) {
-	payments, err := tx.CreateTopLevelBucket(paymentsRootBucket)
-	if err != nil {
-		return nil, err
-	}
+func (p *PaymentControl) nextPaymentSequence() ([]byte, error) {
+	p.paymentSeqMx.Lock()
+	defer p.paymentSeqMx.Unlock()
+
+	// Set a new upper bound in the DB every 1000 payments to avoid
+	// conflicts on the sequence when using etcd.
+	if p.currPaymentSeq == p.storedPaymentSeq {
+		var currPaymentSeq, newUpperBound uint64
+		if err := kvdb.Update(p.db.Backend, func(tx kvdb.RwTx) error {
+			paymentsBucket, err := tx.CreateTopLevelBucket(
+				paymentsRootBucket,
+			)
+			if err != nil {
+				return err
+			}
 
-	seq, err := payments.NextSequence()
-	if err != nil {
-		return nil, err
+			currPaymentSeq = paymentsBucket.Sequence()
+			newUpperBound = currPaymentSeq + 1000
+			return paymentsBucket.SetSequence(newUpperBound)
+		}, func() {}); err != nil {
+			return nil, err
+		}
+
+		// We lazy initialize the the cached currPaymentSeq here in
+		// using the first nextPaymentSequence() call. This if statement
+		// will auto initialize our stored currPaymentSeq, since by
+		// default both this variable and storedPaymentSeq is zero which
```

Comment:
typo: default both this variable and storedPaymentSeq ~is~ are zero

Reply:
done